src/parsers/problem/DimikOJProblemParser.ts

```diff
@@ -0,0 +1,30 @@
+import { Sendable } from '../../models/Sendable';
+import { TaskBuilder } from '../../models/TaskBuilder';
+import { htmlToElement } from '../../utils/dom';
+import { Parser } from '../Parser';
+
+export class DimikOJProblemParser extends Parser {
+  public getMatchPatterns(): string[] {
+    return ['https://dimikoj.com/problems/*'];
+  }
+
+  public async parse(url: string, html: string): Promise<Sendable> {
+    const doc = htmlToElement(html);
+    const task = new TaskBuilder('DimikOJ').setUrl(url);
+
+    // don't use bangla name, it breaks code compilation in cph
+    const urlParts = url.split('/');
+    const problemNumber = urlParts.at(-2);
+    const problemName = urlParts.at(-1).replace('-', ' ').replace(/(?<=(^| ))./g, c => c.toUpperCase());
+    task.setName(`${problemNumber} ${problemName}`);
+
+    const submissionDiv = doc.querySelector('#submission');
+    const codeBlocks = [...submissionDiv.previousElementSibling.querySelectorAll('code')];
+
+    for (let i = 0; i < codeBlocks.length; i += 2) {
```

Comment:
You use `codeBlocks[i + 1]` in the body, so `i < codeBlocks.length` should be `i < codeBlocks.length - 1`.